tests/common/plugins/sanity_check/checks.py

```diff
@@ -154,6 +155,75 @@ def check_dbmemory(dut):
     logger.info("Done checking database memory")
     return check_result
 
+def check_monit(dut):
+    """
+    @summary: Check whether the Monit is running and whether the services which were monitored by Monit are 
+              running correctly or not.
+    """
+    logger.info("Checking status of each Monit entry...")
+    networking_uptime = dut.get_networking_uptime().seconds
+    timeout = max((MONIT_STABILIZE_MAX_TIME - networking_uptime), 0)
+    interval = 20
+    logger.info("networking_uptime = {} seconds, timeout = {} seconds, interval = {} seconds" \
+                .format(networking_uptime, timeout, interval))
+
+    check_result = {"failed": False, "check_item": "Monit services"}
+
+    if timeout == 0:
+        monit_services_status = dut.get_monit_services_status()
+        if not monit_services_status:
+            logger.info("Monit was not running.")
+            check_result["failed"] = True
+            check_result["failed_reason"] = "Monit was not running"
+            logger.info("Checking status of each Monit entry was done!")
+            return check_result
+
+        check_result["services_status"] = {}
+        for service_name, service_info in monit_services_status.items():
+            if ((service_info["service_type"] == "Filesystem" and service_info["service_status"] != "Accessible")
+                or (service_info["service_type"] == "Process" and service_info["service_status"] != "Running")
+                or (service_info["service_type"] == "Program" and service_info["service_status"] != "Status ok")):
+                check_result["failed"] = True
```

Comment:
It's better to refactor these lines (line 182-186) to a method.

Reply:
If we disable monit for some service (such as route_check), then the check will never pass?

Reply:
If we disabled the `route_check` program by the command `sudo monit unmonitor routeCheck`, the monitoring status of `route_check` program will be changed from "Status ok" to "Not monitored". That means, the sanity check for Monit services can not pass.

Reply:
I agree with Bing, "Not monitored" should be a condition that sanity check passes.

Reply:
I also agree with Bing's second update. Sanity check should treat 'Not monitored' as pass. If we need to have a test to restore enable all monitoring and check all monitoring are enabled, we should add a separate test for that.

Reply:
@bingwang-ms I used a separate function `check_monit_service_status(...)`  to check the status of each service. Can you please help me review again?